workspaces/src/network/config.rs

```diff
@@ -0,0 +1,156 @@
+            // key not present in the `into` map. Just insert and move on.
+            into.insert(key, from_val);
+            continue;
+        };
+
+        let val = match (from_val, into_val) {
+            // Overwrite the map recursively:
+            (Value::Object(from), Value::Object(into)) => overwrite_map(from, into)?,
+
+            // Not the same types. One's a map and one's not
+            (value, map @ Value::Object(_)) => {
+                return Err(ErrorKind::DataConversion.message(format!(
+                    "sandbox set config with not the same type: {value:?} vs {map:?}"
+                )))
+            }
+            (map @ Value::Object(_), value) => {
+                return Err(ErrorKind::DataConversion.message(format!(
+                    "sandbox set config with not the same type: {map:?} vs {value:?}"
+                )))
+            }
+
+            // Both are non Map types: just overwrite it.
+            (from, _) => from,
+        };
+        into.insert(key, val);
+    }
+
+    serde_json::to_value(into).map_err(|e| ErrorKind::DataConversion.custom(e))
+}
+
+/// Overwrite the $home_dir/config.json file over a set of entries. `value` will be used per (key, value) pair
+/// where value can also be another dict. This recursively sets all entry in `value` dict to the config
+/// dict, and saves back into `home_dir` at the end of the day.
+fn overwrite(home_dir: &Path, value: Value) -> Result<()> {
+    let config_file =
+        File::open(home_dir.join("config.json")).map_err(|err| ErrorKind::Io.custom(err))?;
+    let config = BufReader::new(config_file);
+    let config: Map<String, Value> =
+        serde_json::from_reader(config).map_err(|err| ErrorKind::DataConversion.custom(err))?;
+
+    let Value::Object(value) = value else {
+        return Err(ErrorKind::DataConversion.message(format!("Setting sandbox's config.json with invalid value: {value:?}")));
+    };
+    let config = overwrite_map(value, config)?;
+    let config_file =
+        File::create(home_dir.join("config.json")).map_err(|err| ErrorKind::Io.custom(err))?;
+    serde_json::to_writer(config_file, &config).map_err(|err| ErrorKind::Io.custom(err))?;
+
+    Ok(())
+}
+
+/// Limit how much nearcore/sandbox can receive per payload. The default set by nearcore is not
+/// enough for certain sandbox operations like patching contract state in the case of contracts
+/// larger than 10mb.
+fn max_sandbox_json_payload_size() -> Result<u64> {
+    let max_files = match std::env::var("NEAR_SANDBOX_MAX_PAYLOAD_SIZE") {
+        Ok(val) => val
+            .parse::<u64>()
+            .map_err(|err| ErrorKind::DataConversion.custom(err))?,
+
+        // Default is 1GB which should suit most contract sizes.
+        Err(_err) => 1024 * 1024 * 1024,
+    };
+
+    Ok(max_files)
+}
+
+/// Set extra configs for the sandbox defined by workspaces.
+pub(crate) fn set_sandbox_configs(home_dir: &Path) -> Result<()> {
+    overwrite(
```

Comment:
Can this be replaced by [json_patch's merge](https://docs.rs/json-patch/0.2.7/json_patch/fn.merge.html) and a bit of IO?